File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/data/repository/CommentsRepositoryImpl.java
package com.example.postscommentsapp.data.repository;

import com.example.postscommentsapp.data.local.dao.CommentDao;
import com.example.postscommentsapp.data.local.entity.CommentEntity;
import com.example.postscommentsapp.data.mapper.CommentMapper;
import com.example.postscommentsapp.domain.model.Comment;
import com.example.postscommentsapp.domain.repository.CommentsRepository;

import java.util.List;

import javax.inject.Inject;

public class CommentsRepositoryImpl implements CommentsRepository {

    private final CommentDao commentDao;

    @Inject
    public CommentsRepositoryImpl(CommentDao commentDao) {
        this.commentDao = commentDao;
    }

    @Override
    public List<Comment> getCommentsByPost(int postId) {
        return CommentMapper.toDomainList(commentDao.getCommentsByPost(postId));
    }

    @Override
    public void insertComment(Comment comment) {
        commentDao.insertComment(
                new CommentEntity(
                        0,
                        comment.getPostId(),
                        comment.getComment(),
                        comment.getCreatedAt()
                )
        );
    }
}

File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/data/repository/CommentsRepositoryImpl.kt
package com.example.postscommentsapp.data.repository

import com.example.postscommentsapp.data.local.dao.CommentDao
import com.example.postscommentsapp.data.local.entity.CommentEntity
import com.example.postscommentsapp.data.mapper.CommentMapper
import com.example.postscommentsapp.domain.model.Comment
import com.example.postscommentsapp.domain.repository.CommentsRepository
import javax.inject.Inject

class CommentsRepositoryImpl @Inject constructor(
    private val commentDao: CommentDao
) : CommentsRepository {

    override suspend fun getCommentsByPost(postId: Int): List<Comment> {
        return CommentMapper.toDomainList(commentDao.getCommentsByPost(postId))
    }

    override suspend fun insertComment(comment: Comment) {
        val entity = CommentEntity(
            id = 0,
            postId = comment.postId,
            comment = comment.comment,
            createdAt = comment.createdAt
        )
        commentDao.insertComment(entity)
    }
}
